File: src/main/java/com/anfema/utils/NetworkUtils.java
package com.anfema.utils;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUtils
{
	public static boolean isConnected( Context context )
	{
		NetworkInfo activeNetworkInfo = getNetworkInfo( context );
		return activeNetworkInfo != null && activeNetworkInfo.isConnected();
	}

	public static boolean isWifiConnected( Context context )
	{
		NetworkInfo activeNetworkInfo = getNetworkInfo( context );
		return activeNetworkInfo != null && activeNetworkInfo.isConnected() && activeNetworkInfo.getTypeName().equalsIgnoreCase( "WIFI" );
	}

	public static NetworkInfo getNetworkInfo( Context context )
	{
		ConnectivityManager connectivityManager = ( ConnectivityManager ) context.getSystemService( Context.CONNECTIVITY_SERVICE );
		return connectivityManager.getActiveNetworkInfo();
	}
}

File: src/main/java/com/anfema/utils/NetworkUtils.kt
package com.anfema.utils

import android.content.Context
import android.net.ConnectivityManager
import android.net.NetworkInfo
import androidx.core.content.getSystemService

object NetworkUtils {

    @JvmStatic
    fun isConnected(context: Context): Boolean =
        getNetworkInfo(context)?.isConnected == true

    private fun getNetworkInfo(context: Context): NetworkInfo? =
        context.getSystemService<ConnectivityManager>()?.activeNetworkInfo
}
